cmd/legacy_param_mapper.go

```diff
@@ -35,6 +35,14 @@ func PopulateNewConfigFromLegacyFlagsAndConfig(c cliContext, flags *flagStorage,
 		return nil, fmt.Errorf("PopulateNewConfigFromLegacyFlagsAndConfig: unexpected nil flags or mount config")
 	}
 
+	// resolve custom endpoint url type to string.
+	var customEndPoint string
+	if flags.CustomEndpoint == nil {
+		customEndPoint = ""
+	} else {
+		customEndPoint = flags.CustomEndpoint.String()
+	}
```

Comment:
This is more succinct.
```suggestion
	var customEndPoint string
	if flags.CustomEndpoint != nil {
		customEndPoint = flags.CustomEndpoint.String()
	} 
```